### PR TITLE
Fixed User as well as added javadoc

### LINKED ISSUE #32: Writing Javadoc for UserDAO, fix problematic parts of User

### PR DESCRIPTION
**Pull request by mazanec202**
Closes #32

### CODE REVIEW

File: src/main/java/tennisclub/entity/User.java [hunk 1]
```diff
@@ -29,61 +31,67 @@
...
+    private Set<Booking> bookings = new HashSet<>();
+
+    @ManyToMany(mappedBy = "teachers")
+    private Set<Lesson>  lessonsToTeach = new HashSet<>();
+
+    @ManyToMany(mappedBy = "students")
+    private Set<Lesson> lessonsToAttend = new HashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private Set<Ranking> rankings = new HashSet<>();
```
Comment (Fuyunori) [inline src/main/java/tennisclub/entity/User.java:44]: Shouldn't be ranking the owning side? 👀 I might be mistaken though 🤔

File: src/main/java/tennisclub/entity/User.java [hunk 2]
```diff
@@ -29,61 +31,67 @@
...
+    public Set<Booking> getBookings() { return bookings; }
+
+    public void addBooking(Booking booking) {
+        bookings.add(booking);
+        booking.addUser(this);
     }
 
+    public Set<Lesson> getLessonsToTeach() { return lessonsToTeach; }
 
+    public void addLessonToTeach(Lesson lesson) {
```
Comment (Fuyunori) [inline src/main/java/tennisclub/entity/User.java:78]: Shouldn't there also be removeLessonToTeach? 👀

File: src/main/java/tennisclub/entity/User.java (hunk 1 above)
Comment (bigno78) [reply src/main/java/tennisclub/entity/User.java:44]: The side that doesn't contain the `mappedBy` is the owning side. So in this case it is indeed Ranking.

Comment (mazanec202) [reply src/main/java/tennisclub/entity/User.java:44]: Yeah, it is slightly confusing as `user` is in this case an attribute of Ranking. However thanks to this I noticed I changed the name of the attribute from `user` to `player` and forgot to change it here :)

File: src/main/java/tennisclub/entity/User.java [hunk 3]
```diff
@@ -29,61 +31,67 @@
...
-        return role;
-    }
+    public Role getRole() { return role; }
+
+    public void setRole(Role role) { this.role = role; }
 
-    public void setRole(Role role) {
-        this.role = role;
+
+    public Set<Booking> getBookings() { return bookings; }
```
[the remark this answers is not in the document]
Comment (mazanec202) [reply src/main/java/tennisclub/entity/User.java:69]: Oh, I wasn't aware of this, thanks!

File: src/main/java/tennisclub/entity/ranking/Ranking.java [hunk 4]
```diff
@@ -25,23 +25,25 @@
     @ManyToOne
     @NotNull
     @Id
-    private User user;
+    private User player;
 
     private int playerPlacement;
 
-    public  Ranking() { }
+    protected Ranking() { }
```
[the remark this answers is not in the document]
Comment (mazanec202) [reply src/main/java/tennisclub/entity/ranking/Ranking.java:32]: Ranking uses compound primary key composed of User and Tournament. This setup requires to have a public/protected constructor without any attributes. As Ranking without a primary key should not exist (both User and Tournament cannot be null), it wouldn't really make sense if a Ranking would be constructed without these attributes.

**Review by bigno78 — APPROVED** [review]

File: src/main/java/tennisclub/entity/User.java [hunk 5]
```diff
@@ -5,17 +5,21 @@
 
 import javax.persistence.*;
 import javax.validation.constraints.NotNull;
+import java.awt.print.Book;
```
Comment (paveltobias) [inline src/main/java/tennisclub/entity/User.java:8]: This probably should not be here.

**Review by paveltobias — APPROVED** [review]

File: src/main/java/tennisclub/entity/User.java (hunk 5 above)
Comment (mazanec202) [reply src/main/java/tennisclub/entity/User.java]: Well spotted, thanks :D

**Pull request merged**